### PR TITLE
Fix stm32f0l0g0 GPIO definitions

### PR DESCRIPTION
**Pull request by takumiando**
## Summary

Fix definitions for GPIOs in stm32f0l0g0 series.

## Impact

Some chips in stm32f0l0g0 series cannot use specific GPIO ports that they have.

## Testing

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

**Comment by takumiando** [comment]
I added👍

File: arch/arm/src/stm32f0l0g0/stm32_gpio.h [hunk 1]
```diff
@@ -186,8 +186,16 @@
...
+#if defined (CONFIG_STM32F0L0G0_STM32F03X)
+#  define GPIO_PORTF                  (4 << GPIO_PORT_SHIFT)     /*   GPIOF */
+#else
 #  define GPIO_PORTE                  (4 << GPIO_PORT_SHIFT)     /*   GPIOE */
+#if defined (CONFIG_ARCH_CHIP_STM32L0)
+#  define GPIO_PORTH                  (5 << GPIO_PORT_SHIFT)     /*   GPIOH */
+#else
 #  define GPIO_PORTF                  (5 << GPIO_PORT_SHIFT)     /*   GPIOF */
+#endif
+#endif
```
Comment (raiden00pl) [inline arch/arm/src/stm32f0l0g0/stm32_gpio.h:198]: Why just not add GPIO_PORTH definition here ? When a chip doesn't support a given GPIO, then it simply doesn't use a given value. I see no point in complicating this logic.

```suggestion
#  define GPIO_PORTE                  (4 << GPIO_PORT_SHIFT)     /*   GPIOE */
#  define GPIO_PORTF                  (5 << GPIO_PORT_SHIFT)     /*   GPIOF */
#  define GPIO_PORTH                  (6 << GPIO_PORT_SHIFT)     /*   GPIOH */
```

Comment (raiden00pl) [reply arch/arm/src/stm32f0l0g0/stm32_gpio.h:198]: Never mind. This value is directly used to get base address for GPIO block, so this conditional logic is necessary.

Comment (takumiando) [reply arch/arm/src/stm32f0l0g0/stm32_gpio.h:198]: Yes, ```GPIO_PORTx``` is referred as a index of base addresses in ```arch/arm/src/stm32f0l0g0/stm32_gpio.c```.

```c
const uint32_t g_gpiobase[STM32_NPORTS] =
{
#if STM32_NPORTS > 0
  STM32_GPIOA_BASE,   /* One GPIO ports, GPIOA */
#endif
#if STM32_NPORTS > 1
  STM32_GPIOB_BASE,   /* Two GPIO ports, GPIOA-B */
#endif
#if STM32_NPORTS > 2
  STM32_GPIOC_BASE,   /* Three GPIO ports, GPIOA-C */
#endif
#if STM32_NPORTS > 3
  STM32_GPIOD_BASE,   /* Four GPIO ports, GPIOA-D */
#endif
#if defined(STM32_GPIOE_BASE)
  STM32_GPIOE_BASE,   /* GPIOE */
#endif
#if defined(STM32_GPIOF_BASE)
  STM32_GPIOF_BASE,   /* GPIOF */
#endif
#if defined(STM32_GPIOH_BASE)
  STM32_GPIOH_BASE,   /* GPIOH */
#endif
};
```

**Pull request merged**